python/ray/dashboard/client/src/pages/dashboard/tune/TuneTable.tsx

```diff
@@ -0,0 +1,235 @@
+import { Theme } from "@material-ui/core/styles/createMuiTheme";
+import createStyles from "@material-ui/core/styles/createStyles";
+import withStyles, { WithStyles } from "@material-ui/core/styles/withStyles";
+import React from "react";
+import { connect } from "react-redux";
+import { StoreState } from "../../../store";
+import { dashboardActions } from "../state";
+import Table from "@material-ui/core/Table";
+import TableBody from "@material-ui/core/TableBody";
+import TableCell from "@material-ui/core/TableCell";
+import TableHead from "@material-ui/core/TableHead";
+import TableRow from "@material-ui/core/TableRow";
+import TableSortLabel from "@material-ui/core/TableSortLabel";
+import { TuneTrial } from "../../../api";
+
+const styles = (theme: Theme) =>
+  createStyles({
+    root: {
+      padding: theme.spacing(2),
+      "& > :not(:first-child)": {
+        marginTop: theme.spacing(2)
+      }
+    },
+    table: {
+      marginTop: theme.spacing(1)
+    },
+    cell: {
+      padding: theme.spacing(1),
+      textAlign: "right",
+      "&:last-child": {
+        paddingRight: theme.spacing(1)
+      }
+    }
+  });
+
+const mapStateToProps = (state: StoreState) => ({
+  tuneInfo: state.dashboard.tuneInfo
+});
+
+interface State {
+  metricParamColumn: string;
+  ascending: boolean;
+  sortedColumn: keyof TuneTrial | undefined;
+}
+
+const mapDispatchToProps = dashboardActions;
+
+class TuneTable extends React.Component<
+  WithStyles<typeof styles> &
+    ReturnType<typeof mapStateToProps> &
+    typeof mapDispatchToProps,
+  State
+> {
+  timeout: number = 0;
+
+  state: State = {
+    sortedColumn: undefined,
+    ascending: true,
+    metricParamColumn: ""
+  };
+
+  onColumnClick = (column: keyof TuneTrial, metricParamColumn?: string) => {
+    let ascending = this.state.ascending;
+    if (column === this.state.sortedColumn) {
+      ascending = !ascending;
+    } else {
+      ascending = true;
+    }
+    this.setState({
+      sortedColumn: column,
+      ascending: ascending
+    });
+
+    if (metricParamColumn) {
+      this.setState({
+        metricParamColumn: metricParamColumn
+      });
+    }
+  };
+
+  humanize = (str: string) => {
+    var i,
+      frags = str.split("_");
+    for (i = 0; i < frags.length; i++) {
+      frags[i] = frags[i].charAt(0).toUpperCase() + frags[i].slice(1);
+    }
+    return frags.join(" ");
+  };
+
+  sortedCell = (name: keyof TuneTrial, mPC?: string) => {
```

Comment:
please make sure to fully spell out `mPC` or rename it, otherwise the code can be hard to maintain